### PR TITLE
Refactor Tile Inspector

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/openrct2/world/TileInspector.cpp [hunk 1]
```diff
@@ -30,201 +30,201 @@
...
 
-        tile_element_remove(tileElement);
-        map_invalidate_tile_full(loc);
+            tile_element_remove(tileElement);
+            map_invalidate_tile_full(loc);
 
-        // Update the window
-        rct_window* const tileInspectorWindow = window_find_by_class(WC_TILE_INSPECTOR);
-        if (tileInspectorWindow != nullptr && loc == windowTileInspectorTile.ToCoordsXY())
-        {
```
Comment (Broxzier) [inline src/openrct2/world/TileInspector.cpp:227]: It looks like that now this check is removed, the code bellow will be called for anyone, even if they have a different tile selected.

File: src/openrct2/world/TileInspector.cpp [hunk 2]
```diff
@@ -236,164 +236,154 @@ GameActionResultPtr tile_inspector_remove_element_at(const CoordsXY& loc, int16_
...
+            if (!SwapTileElements(loc, first, second))
+            {
+                return std::make_unique<GameActions::Result>(GameActions::Status::Unknown, STR_NONE);
+            }
+            map_invalidate_tile_full(loc);
 
-        // Update the window
-        rct_window* const tileInspectorWindow = window_find_by_class(WC_TILE_INSPECTOR);
-        if (tileInspectorWindow != nullptr && loc == windowTileInspectorTile.ToCoordsXY())
-        {
```
Comment (Broxzier) [inline src/openrct2/world/TileInspector.cpp:259]: Same here

File: src/openrct2/world/TileInspector.cpp [hunk 3]
```diff
@@ -30,1162 +30,1104 @@
...
 
-        // Set the base height to be the same as the selected element
-        TileElement* const selectedElement = map_get_nth_element_at(loc, elementIndex + 1);
-        if (selectedElement == nullptr)
+        return true;
+    }
+
+    static rct_window* GetTileInspectorWithPos(const CoordsXY& loc)
+    {
+        // Update the tile inspector's list for everyone who has the tile selected
```
Comment (Broxzier) [inline src/openrct2/world/TileInspector.cpp:78]:
```suggestion
        // Return the tile inspector window for everyone who has the tile selected
```

File: src/openrct2/world/TileInspector.cpp [hunk 4]
```diff
@@ -30,1162 +30,1104 @@
...
-        const auto rotatedCurrentTile = CoordsXYZ{ CoordsXY{ tiles[i].x_offset, tiles[i].y_offset }.Rotate(direction),
-                                                   tiles[i].z_offset };
+        const rct_scenery_entry* const largeEntry = largeScenery->GetEntry();
+        const auto* const tiles = largeEntry->large_scenery.tiles;
+        const auto direction = largeScenery->GetDirection();
+
+        const auto rotatedFirstTile = CoordsXYZ{ CoordsXY{ tiles[largeScenery->GetSequenceIndex()].x_offset,
+                                                           tiles[largeScenery->GetSequenceIndex()].y_offset }
+                                                     .Rotate(direction),
+                                                 tiles[largeScenery->GetSequenceIndex()].z_offset };
```
Comment (Broxzier) [inline src/openrct2/world/TileInspector.cpp:159]: A trailing comma will likely make this readable:

```suggestion
        const auto rotatedFirstTile = CoordsXYZ{
            CoordsXY{ tiles[largeScenery->GetSequenceIndex()].x_offset, tiles[largeScenery->GetSequenceIndex()].y_offset }
                .Rotate(direction),
            tiles[largeScenery->GetSequenceIndex()].z_offset,
        };
```

**Review by Broxzier — CHANGES_REQUESTED** [review]
The changes look good. I've tested it with a server and client.

File: src/openrct2/world/TileInspector.cpp (hunk 4 above)
Comment (ZehMatt) [reply src/openrct2/world/TileInspector.cpp:159]: I can probably do a little better by having sequence index being a separate variable also.

Comment (ZehMatt) [reply src/openrct2/world/TileInspector.cpp]: Still pretty hefty but a bit better now.

**Review by Broxzier — APPROVED** [review]

**Pull request merged**